fuzz/db_fuzzer.cc

```diff
@@ -0,0 +1,135 @@
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <filesystem>
+
+#include "rocksdb/db.h"
+
+// Helper function to getting strings based on the random data.
+std::string get_string(const uint8_t** data, size_t* remaining) {
+  if (*remaining == 0) {
+    return std::string("");
+  }
+  uint8_t s_size = (*data)[0];
+
+  // if s_size is more than we have available, do not use s_size
+  if (s_size > *remaining) {
+    std::string s1(reinterpret_cast<const char*>(*data), *remaining);
+    *data += *remaining;
+    *remaining = 0;
+    return s1;
+  }
+
+  std::string s1(reinterpret_cast<const char*>(*data), s_size);
+  *remaining -= s_size;
+  return s1;
+}
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
```

Comment:
Can we use [structure-aware fuzzing](https://github.com/google/fuzzing/blob/master/docs/structure-aware-fuzzing.md) instead of relying on custom interpretation of the `data` byte array?
Maybe we can extend `fuzz/db_operation.proto` to include more operations?

Reply:
the use of  `FuzzedDataProvider` make the logic clearer!